File: check_pangram_sentence.cpp
#include "leetcode.h"

class Solution {
public:
  bool checkIfPangram(string sentence) {
    set<int> s;
    for (auto &i : sentence)
      s.insert(i);
    return s.size() == 26;
  }
};

int main() {
  Solution obj;
  if (obj.checkIfPangram("thequickbrownfoxjumpsoverthelazydog"))
    cout << "true";
  else
    cout << "false";
}

File: other/check_pangram_sentence.rs
struct Solution;

impl Solution {
    pub fn check_if_pangram(sentence: String) -> bool {
        let mut bit_mask: u32 = 0;
        let mut count = 0;
        for c in sentence.chars() {
            let idx = c as usize - 'a' as usize;
            if bit_mask & (1 << idx) == 0 {
                count += 1;
            }
            bit_mask |= 1 << idx;
        }
        count == 26
    }
}

fn main() {
    let sentence = String::from("thequickbrownfoxjumpsoverthelazydog");
    print!("{}", Solution::check_if_pangram(sentence));
}
